docs/user-guides/retrieval.md

```diff
@@ -0,0 +1,282 @@
+# Retrieval in CLIP-as-service
+`CLIP-as-service` offers us high-quality embeddings. Retrieval is one of the most common use cases for embeddings. Retrieval in `CLIP-as-service` support indexing a very large dataset (millions/billions) and querying within 50ms, depending on the machine.
+
+In order to implement retrieval, we add an [`AnnLite`](https://github.com/jina-ai/annlite) indexer executor after the encoder executor in CLIP-as-service.
```

Comment:
It would be nice to give a high-level overview of how the retrieval works/which steps make up the process -> the part from: How Jina eco-system makes cross-modality simple and efficient